File: tfliteapp/app/src/main/java/com/example/tfliteapp/MainActivity.java
package com.example.tfliteapp;

import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.media.ThumbnailUtils;
import android.os.Build;
import android.os.Bundle;
import android.provider.MediaStore;
import android.service.autofill.FieldClassification;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.tfliteapp.ml.BirdsModel;

import org.tensorflow.lite.support.image.TensorImage;
import org.tensorflow.lite.support.label.Category;

import java.io.IOException;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    TextView result, confidence;
    ImageView imageView;
    Button picture;
    int imageSize = 224;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        result = findViewById(R.id.result);
        confidence = findViewById(R.id.confidence);
        imageView = findViewById(R.id.imageView);
        picture = findViewById(R.id.button);

        picture.setOnClickListener(new View.OnClickListener() {
            @RequiresApi(api = Build.VERSION_CODES.M)
            @Override
            public void onClick(View view) {
                // Launch camera if we have permission
                if (checkSelfPermission(Manifest.permission.CAMERA) == PackageManager.PERMISSION_GRANTED) {
                    Intent cameraIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
                    startActivityForResult(cameraIntent, 1);
                } else {
                    //Request camera permission if we don't have it.
                    requestPermissions(new String[]{Manifest.permission.CAMERA}, 100);
                }
            }
        });
    }

    public void classifyImage(Bitmap img)
         {
             try {
                 BirdsModel model = BirdsModel.newInstance(getApplicationContext());

                 // Creates inputs for reference.
                 TensorImage image = TensorImage.fromBitmap(img);

                 // Runs model inference and gets result.
                 BirdsModel.Outputs outputs = model.process(image);
                 List<Category> probability = outputs.getProbabilityAsCategoryList();

                 // Releases model resources if no longer used.
                 model.close();
             } catch (IOException e) {
                 // TODO Handle the exception
             }
         }

    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if (requestCode == 1 && resultCode == RESULT_OK) {
             Bitmap image=(Bitmap) data.getExtras().get("data");
             int dimension= Math.min(image.getWidth(),image.getHeight());
             image=ThumbnailUtils.extractThumbnail(image,dimension,dimension);
             imageView.setImageBitmap(image);

             image=Bitmap.createScaledBitmap(image,imageSize,imageSize,false);
             classifyImage(image);

        }
        super.onActivityResult(requestCode, resultCode, data);
    }
}

File: app/src/main/java/com/example/image_classification_app/MainActivity.kt
package com.example.image_classification_app

import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.content.ContextCompat
import androidx.core.content.PackageManagerCompat
import com.example.image_classification_app.databinding.ActivityMainBinding


    get() {}

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var imageView: ImageView
    private lateinit var button: Button
    private lateinit var tvOutput: TextView
    private val GALLERY_REQUEST_CODE=123


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root

        setContentView(view)

        imageView=binding.imageView
        button=binding.binCaptureImage
        tvOutput=binding.textview
        val buttonLoad=binding.binLoadImage

        button.setOnClickListener {
            if (ContextCompat.checkSelfPermission(this, android.Manifest.permission.CAMERA)
                == PackageManager.PERMISSION_GRANTED)
                {
                 textPicturePreview.launch(null)
                }
            else
            {
             requestPermission.launch(android.Manifest.permission.CAMERA)
            }
        }
        
    }
    // Launch camera and take a picture
    private val takePicturePreview= registerForActivityResult(ActivityResultContracts.TakePicturePreview()){
        bitmap->if(bitmap!=null)
    {
     imageView.setImageBitmap(bitmap)
     outputGenerator(bitmap)

    }
    }
}